services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py

```diff
@@ -469,23 +470,55 @@ async def get_project_services_access_for_gid(request: web.Request) -> web.Respo
         ]
     )
 
+    # Initialize groups to compare with everyone group ID
+    groups_to_compare = {EVERYONE_GROUP_ID}
+
+    # Get the group from the provided group ID
+    _sharing_with_group: Group | None = await get_group_from_gid(
+        app=request.app, gid=query_params.for_gid
+    )
+
+    # Check if the group exists
+    if _sharing_with_group is None:
+        msg = f"Group ID does not exist: {query_params.for_gid}"
+        raise ValueError(msg)
+
+    # Update groups to compare based on the type of sharing group
+    if _sharing_with_group.group_type == GroupTypeInModel.PRIMARY:
+        _user_id = await get_user_id_from_gid(
+            app=request.app, primary_gid=query_params.for_gid
+        )
+        _, _user_groups, _ = await list_user_groups(app=request.app, user_id=_user_id)
+        groups_to_compare.update({int(item.get("gid")) for item in _user_groups})
+        groups_to_compare.add(query_params.for_gid)
+    elif _sharing_with_group.group_type == GroupTypeInModel.STANDARD:
+        groups_to_compare = {query_params.for_gid}
```

Comment:
So the idea is that for `gid` that are primary, the access rights is checked against ALL user's groups but if `gid` is standard then access rights are only checked against it.

I understand the underline problem that solves but it also feels that we are mixing responsibilities between front-end and backend here. Let me try to give you some examples:

- If the primary gid is special, what about simply having the query option to `for_user_id` that acts as you say?
- Why the front-end needs the primary `gid` for? shouldn't it only solely rely on the `user_id` to refer to a user? 
- Same happens with products, that also have a group associated.
- IMO groups is a concept of the backend. The only group the front-end should know about is what we call "organizations" and perhaps the entire API should be build on taht bases, i.e. we have "users", "products" and "organizations" ... but not groups

Reply:
Yes, you got the idea right! :+1: I will respond in two sections: 1. practical and 2. discussion.

1. practical
- With this PR, I have not introduced any new function or frontend interface. I am only reusing what is already there with the best intentions to solve the problem users were complaining about: the warning was popping up even though they already had access to the service. I even improved the group plugin a bit so that we do not expose RowProxy outside of the DB layer.

2. discussion
- I fully agree with you that groups should not have been exposed to the frontend; it should have been an implementation detail on our side. I also think we are missing an Organization resource. Currently, a subset of groups equals an organization, and this is the confusing part. That is probably why the group type was introduced, which I think is also not fully necessary.
- But still, I do not think it is too bad. Basically, we just need to accept (unless we refactor it) that in most cases, groups equal organizations. The frontend is already working with it like that, and yes, I agree the API might have been more straightforward with resources like "users," "products," and "organizations." However, on the other hand, if we accept that groups equal organizations, then for_gid also makes sense. In the frontend, when sharing, you can see a list of organizations (including users as an organization with one member). And you are sharing the project with an organization (an organization can have just one member ;)).
- I guess we can discuss the design and how we can improve and whether we want/need to improve this part of the code. But back to the practical aspect - the purpose of this PR is only to improve the current situation without introducing something new.

Reply:
Just to add some noise here.
Groups predate any organization or even products I think. Products were not linked to groups before late.
Also Groups are not equal to organizations. Organizations are contained inside Groups or please define what an organization is. For me a group can be any group of users, sub groups inside an organization or cross-organizations even like the everyone group. Groups were originally created very similarly to the groups in the linux OS which follows the same thing (primary, standard, everyone).
I don't really get why the groups should not have been exposed to the frontend.??

Reply:
Also originally there were no project-products ownerships... this would all only work through the services. so I guess there might be leeway also to improve that whole thing.

Reply:
This is not the first time we discuss about this :-) 
I simply have the feeling we are misusing groups most like likely because we interprete the differently. 

@sanderegg   I totally get your original design. It provides a nice low level abstraction but now, when the groups adquire a meaning (i.e. not just a set with users but e.g. products, organizations, etc) we find use cases that cannot be covered without update the groups idea and adding additional constraints ...

@matusdrobuliak66 I did not mean to say that groups has to be equal to organizations. This is how the front-end is using them. 

I guess it would be better to do this discussion offline. i will try to prepare some ideas for next time we meet.

Reply:
Yes agree we should discuss in person and create some issue with actions based on that :+1: